Source/ProgressionSystemRuntime/Private/Components/PSHUDComponent.cpp

```diff
@@ -0,0 +1,150 @@
+﻿// Copyright (c) Valerii Rotermel & Yevhenii Selivanov
+
+#include "Components/PSHUDComponent.h"
+//---
+#include "Bomber.h"
+#include "Controllers/MyPlayerController.h"
+#include "UtilityLibraries/MyBlueprintFunctionLibrary.h"
+#include "Data/PSDataAsset.h"
+#include "Blueprint/WidgetTree.h"
+#include "UI/MyHUD.h"
+#include "Widgets/PSMenuWidget.h"
+#include "Data/PSTypes.h"
+#include "Data/PSSaveGameData.h"
+#include "Data/PSWorldSubsystem.h"
+#include "Components/Image.h"
+#include "GameFramework/MyGameStateBase.h"
+#include "GameFramework/MyPlayerState.h"
+#include "MyDataTable/MyDataTable.h"
+#include "MyUtilsLibraries/WidgetUtilsLibrary.h"
+#include "Subsystems/GlobalEventsSubsystem.h"
+
+#include UE_INLINE_GENERATED_CPP_BY_NAME(PSHUDComponent)
+
+// Sets default values for this component's properties
+UPSHUDComponent::UPSHUDComponent()
+{
+	// Set this component to be initialized when the game starts, and to be ticked every frame.  You can turn these features
+	// off to improve performance if you don't need them.
+	PrimaryComponentTick.bCanEverTick = false;
+	PrimaryComponentTick.bStartWithTickEnabled = false;
+}
+
+// Subscribes to the end game state change notification on the player state.
+void UPSHUDComponent::OnLocalPlayerStateReady(AMyPlayerState* PlayerState, int32 CharacterID)
+{
+	// Ensure that PlayerState is not null before subscribing to the event
+	checkf(PlayerState, TEXT("ERROR: 'PlayerState' is null"));
+	PlayerState->OnEndGameStateChanged.AddUniqueDynamic(this, &ThisClass::OnEndGameStateChanged);
+}
+
+// Called when the game starts
+void UPSHUDComponent::BeginPlay()
+{
+	Super::BeginPlay();
+
+	AMyHUD* MyHUD = CastChecked<AMyHUD>(GetOwner());
+	const AMyHUD& HUD = *MyHUD;
+	ProgressionMenuWidgetInternal = HUD.CreateWidgetByClass<UPSMenuWidget>(UPSDataAsset::Get().GetProgressionMenuWidget(), true, 1);
```

Comment:
Please sync to latest and fix compile error related to `MyHUD` in `UPSHUDComponent`. Now `CreateWidget` function is in `FWidgetUtilsLibrary` instead of `MyHUD`.

Perform next steps to make it compile and work again:
1 - Replace next code from:
```c++
  AMyHUD* MyHUD = CastChecked<AMyHUD>(GetOwner());
  const AMyHUD& HUD = *MyHUD;
  ProgressionMenuWidgetInternal = HUD.CreateWidgetByClass<UPSMenuWidget>(UPSDataAsset::Get().GetProgressionMenuWidget(), true, 1);
  checkf(ProgressionMenuWidgetInternal, TEXT("ERROR: 'ProgressionMenuWidgetInternal' is null"));
```
by:
```c++
ProgressionMenuWidgetInternal = FWidgetUtilsLibrary::CreateWidgetChecked<UPSMenuWidget>(UPSDataAsset::Get().GetProgressionMenuWidget(), true, 1);
```
_Note: you don't need anymore `checkf` after `CreateWidgetChecked` as it's already `Checked` inside_.

2 - In engine, in the root of your module, open `ProgressionSystem` MGF data asset, find `PSHUDComponent`, and for `Actor Class` property select `MyGameStateBase` (before it was `MyHUD`):
![MGF](https://github.com/h4rdmol/ProgressionSystem/assets/20540872/62fc0c43-9c18-4ff6-91ac-a2a2a0ea7e72)

3 - Before submit, please retest it on [**CLIENT**] multiplayer

Reply:
resolved as part of https://github.com/h4rdmol/ProgressionSystem/pull/1/commits/aaf19939eb4c1582f79acbe02292616d28ec83df

Reply:
@h4rdmol Is not finished.

1. You again ignored next from above, please remove checkf 😫
| Note: you don't need anymore checkf after CreateWidgetChecked as it's already Checked inside.

2. Your `MyMenuWidget` local variable does not make any sense, just set `ProgressionMenuWidgetInternal` directly.

3. Remove extra space before Super::BeginPlay that you populated with your commit 

![image](https://github.com/h4rdmol/ProgressionSystem/assets/20540872/38fc088d-96c6-44c5-a706-91f56e8dda23)

Reply:
solved https://github.com/h4rdmol/ProgressionSystem/pull/1/commits/2d7b128ccf957a30a145cbcac6097d99177cbc3c
my bad, sorry